test/compilable/testlambdacomp.d

```diff
@@ -132,10 +133,75 @@ void test4()
     bar!((a, b) => a < b + 7);
 }
 
+int bar()
+{
+    return 2;
+}
+
+void foobar2(alias pred)()
+{
+    // imports.testalambda1.bar != imports.testlambda2.bar
+    import imports.testlambda2 : bar;
+    static assert(!__traits(isSame, pred, (int a) => a + bar()));
+}
+
+void foobar(alias pred)()
+{
+    // testlambdacomp.bar != testlambdacomp.test5.bar
+    static assert(!__traits(isSame, pred, (int a) => a + bar()));
+
+    // imports.testlambda1.bar != testlambdacomp.test5.bar
+    import imports.testlambda1 : bar;
+    static assert(!__traits(isSame, pred, (int a) => a + bar()));
+
+    // functions imported from different modules are not equal
+    foobar2!((int a) => a + bar())();
+}
```

Comment:
Ehm that's hard to read - I suggest the following:

- better names than `foobar` and `foobar2`
- put the test that calls `foobar` etc directly below their declaration (maybe even put the methods in the function itself)

Reply:
Alternatively, simply don't declare a function which you just call once, but call it e.g. `test6` and do `alias pred = (int a) => a + bar()` in the beginning.